libs/components/src/lib/switch/switch.ts

```diff
@@ -31,4 +32,78 @@ export class Switch extends FoundationSwitch {
 	 * HTML Attribute: connotation
 	 */
 	@attr connotation?: SwitchConnotation;
+
+	/**
+	 * When true, the control will be immutable by user interaction. See {@link https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/readonly | readonly HTML attribute} for more information.
+	 * @public
+	 * @remarks
+	 * HTML Attribute: readonly
+	 */
+	// @ts-expect-error Type is incorrectly non-optional
+	@attr({ attribute: 'readonly', mode: 'boolean' }) readOnly: boolean; // Map to proxy element
+	// @ts-expect-error Function is declared but never read
+	private readOnlyChanged(): void {
```

Comment:
For consistency with our codebase where we do:
```suggestion
	/**
	 * @internal
	 */
	readOnlyChanged(): void {
```